api/base/utils.py

```diff
@@ -75,8 +75,9 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     return url
 
 
-def get_object_or_error(model_cls, query_or_pk, display_name=None):
+def get_object_or_error(model_cls, query_or_pk, request, display_name=None):
     obj = query = None
+    select_for_update = bool(request.method != 'GET')
```

Comment:
This will not account for `HEAD` and `OPTIONS` requests. Rather than comparing to `GET`, you can use `rest_framework.permissions.SAFE_METHODS` which is a list of read-only methods.